### PR TITLE
INNOVATION TICKET: Add an AI Powered Mock Interview bot (Proof of Concept)

### PR DESCRIPTION
**Pull request by ntotten-kainos**
Added a PoC/Demo for an openAI API Powered mock interview bot.
The bot takes in the task of performing the duties of an interviewer for a given job description. 
The job description is passed to the bot to begin the interview with tailored questions to that role.

In a real deployment this would be tied to the job application page, so a user can practice interviews for a role they are interested in.

Need to manually navigate to <host>:3000/mockInterview to access the bot as we do not have a job application page yet.

### CODE REVIEW

**Review by rackie-pascua — DISMISSED** [review]
looks good to me great work!

File: views/mockInterview.html [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+              <input type="text" id="user-input" placeholder="Type your answer here..." />
+              <button class="button2" onclick="sendAnswer()">Send</button>
+            </div>
+          </div>
+    </div>
+  <script>
+    const chatBox = document.getElementById('chat-box');
+    const userInput = document.getElementById('user-input');
+
+    async function sendAnswer() {
```
Comment (boydstepkainos) [inline views/mockInterview.html:50]: Lets pull this out into a script and make use of it here instead

Comment (ntotten-kainos) [reply views/mockInterview.html]: Had trouble getting it to work nicely with compiled TS > JS so hoping it's ok if I just have a JS script that I link here

**Review by boydstepkainos — APPROVED** [review]

**Review by JamieMcConnellKainos — APPROVED** [review]
All looks good

**Pull request merged**